contributors/design-proposals/bulk_watch.md

```diff
@@ -0,0 +1,450 @@
+# Background
+
+ As part of increasing security of a cluster, we are planning to limit the
+ability of a given Kubelet (in general: node), to be able to read only
+resources associated with it. Those resources, in particular means: secrets,
+configmaps & persistentvolumeclaims. This is needed to avoid situation when
+compromising node de facto means compromising a cluster. For more details &
+discussions see https://github.com/kubernetes/kubernetes/issues/40476.
+ 
+ However, by some extension to this effort, we would like to improve scalability
+of the system, by significantly reducing amount of api calls coming from
+kubelets. As of now, to avoid situation that kubelet is watching all secrets/
+configmaps/... in the system, it is not using watch for this purpose. Instead of
+that, it is retrieving indidual objects, by sending individual GET requests.
+However, to enable automatic updates of mounted secrets/configmaps/..., Kubelet
+is sending those GET requests periodically. In large clusters, this is
+generating huge unnecessary load, as this load in principle should be
+watch-based. We would like to address this together with solving the
+authorization issue.
+
+
+# Proposal
+
+ In this proposal, we're not focusing on how exactly security should be done.
+We're just sketching very high level approach and exact authorization mechanism
+should be discussed separately.
+
+ At the high level, what we would like to achieve is to enable LIST and WATCH
+requests to support more sophisticated filtering (e.g. we would like to be able
+to somehow ask for all Secrets attached to all pods bound to a given node in a
+single LIST or WATCH request). However, the design has to be consistent with
+authorization of other types of requests (in particular GETs).
+
+ To solve this problem, we propose to introduce the idea of `bulk watch ` and
+`bulk list ` (and in the future other bulk operations). This idea was already
+appearing in the community and now we have good usecase to proceed with it.
+
+ Once a bulk watch is set up, we also need to periodically verify its ACLs.
+Whenever a user (in our case Kubelet) loses access to a given resource, the
+watch should be closed within some bounded time. the rationaly behind this
+requirement is that by using (bulk) watch we can't regress the current behavior
```

Comment:
nit: I think this behavior is motivated just by enforcing meaningful acls and there's no need to invoke the current behavior of get as justification.